sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -66,10 +68,23 @@ sealed trait TimestampFormatter extends Serializable {
   @throws(classOf[DateTimeParseException])
   @throws(classOf[DateTimeException])
   @throws(classOf[IllegalStateException])
-  def parseWithoutTimeZone(s: String): Long =
+  def parseWithoutTimeZone(s: String, failOnError: Boolean): Long =
```

Comment:
ditto, `allowTimeZone`?

Reply:
Another question is, instead of adding a new parameter, can the caller side pick one of `parse` and `parseWithoutTimeZone`?

Reply:
I am not sure we can do it, because `parseWithoutTimeZone` method can be used with different options depending on the context, and it is, in fact.